tools/uf2conv.py

```diff
@@ -248,21 +265,12 @@ def get_drives():
             if len(words) >= 3 and words[1] == "2" and words[2] == "FAT":
                 drives.append(words[0])
     else:
-        rootpath = "/run/media"
-        if not os.path.isdir(rootpath):
-            rootpath = "/media"
-        if not os.path.isdir(rootpath):
-            rootpath = "/opt/media"
         if sys.platform == "darwin":
-            rootpath = "/Volumes"
+            possibly_anydir("/Volumes", drives)
         elif sys.platform == "linux":
-            tmp = rootpath + "/" + os.environ["USER"]
-            if os.path.isdir(tmp):
-                rootpath = tmp
-        for d in os.listdir(rootpath):
-            drives.append(os.path.join(rootpath, d))
-
-        if (len(drives) == 0) and (sys.platform == "linux"):
+            # Do first since a race condition is possible.
+            time.sleep(2)
+            # GUI desktop is required (X.org) that supports freedesktop.
```

Comment:
What does this comment mean?  The `/dev` tree is populated by the kernel AIUI...

Reply:
Yes, `/dev` tree is, but the mounting daemon is controlled by freedesktop/udisks rules. But you don't have to have it to do mounts. 
If you run headless, or text only it makes no sense to have udisks... 
and for even more fun...
you can even have udev do the mounting direct without udisks, or manually via sudo/su/root.
It's not a matter of `/dev` entry being created, it's the mounting of the device.